packages/mars/src/values/future.ts

```diff
@@ -40,3 +40,9 @@ export class Future<T> {
     throw new Error('This will get overridden.')
   }
 }
+
+type FutureArray<T> = T extends [V, ...Rest] ? [Future<V>, ...FutureArray<Rest>] : []
```

Comment:
Does not compiler. Did you forget `infer` of sth?

Reply:
`type FutureArray<T> = T extends [infer V, ...(infer Rest)] ? [Future<V>, ...FutureArray<Rest>] : []`